src/Microsoft.AspNet.Razor/CodeGenerators/CSharpCodeWriter.cs

```diff
@@ -470,23 +470,6 @@ private void WriteCStyleStringLiteral(string literal)
                         Write(literal[i].ToString());
                         break;
                 }
```

Comment:
Suggest also removing `literal.Length >= 256 && literal.Length <= 1500` from [line 118](https://github.com/aspnet/Razor/blob/nimullen/stringconcats.614/src/Microsoft.AspNet.Razor/CodeGenerators/CSharpCodeWriter.cs#L118). A verbatim `string` is fine for anything lacking a `'\0'` character.